### ISSUE TITLE
To minimize sum of losses from 2 or more different simple Neural Nets with same architecture.

### ISSUE THREAD

**Issue by akaysh** [denizyuret/Knet.jl#143]
I have a problem where I need to minimize the sum of losses from two or more different Neural Nets. All of them are trained on the same data points.
Loss function for NNs are defined by me and are dependent on the output of all other NNs. Will Knet's `grad` and `update` work in this case?

The approach I followed is like this 

- Made an Array which has parameters for all NNs. For example `Params = [[NN1 params],[NN2 params],...]`

- Passed this array in `Autograd.grad()` to calculate the gradient of output wrt parameters.

- Update the Params according to the gradient and the optimization algo (used Adam)

But it does not seem to work and learn properly. Any suggestions?

**Comment by akaysh** [comment]
Add help required label to this.

**Comment by ozanarkancan** [comment]
Hi,

There shouldn't be any problem. There might be an optimization problem. Are your losses at the same scale (one might be regression, other one might be negative loglikelihood)? Could you share a code snippet?

**Comment by akaysh** [comment]
@ozanarkancan sure.
Here is the link to my repo
https://github.com/JuliaDiffEq/NeuralNetDiffEq.jl

**Comment by akaysh** [comment]
Look at MultiNN branch snippet.
